### PR TITLE
Use built-in CMake support for iOS

### PR DESCRIPTION
**Pull request by JonnyPtn**
Now we're using more modern cmake, iOS support is first-class so we don't need to ship our own toolchain.

Practically there should be no difference except for generally being easier to work with iOS (e.g don't need to reconfigure to switch between simulator/device)

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: cmake/Config.cmake
Comment (eXpl0it3r) [inline cmake/Config.cmake:49]: The comment says "our extlibs folder too", but then the option is set to `NEVER`, doesn't that mean, we're exclusively searching the `extlibs` folder?

File: .github/workflows/ci.yml
Comment (ChrisThrasher) [inline .github/workflows/ci.yml:41]:
```suggestion
          config: { name: iOS, flags: -DCMAKE_SYSTEM_NAME=iOS }
```

File: CMakeLists.txt
Comment (ChrisThrasher) [inline CMakeLists.txt:15]: Can we move this somewhere after the `project` call? I'd like to minimum how much code we add before `project()`.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
I'm not familiar with CMake for iOS but if this works then it looks like a great simplification.

File: CMakeLists.txt
Comment (JonnyPtn) [reply CMakeLists.txt:15]: Nope it has to be before the first call to project https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: cmake/Config.cmake
Comment (JonnyPtn) [reply cmake/Config.cmake:49]: Yeah good spot - originally set to BOTH but then changed to NEVER, will update comment

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML-Buildbot/blob/0ab146c329327a85a191f716e8de41c8070f7325/buildfactories.py#L67-L69

Looks like we need to update the Buildbot too

**Comment by JonnyPtn** [comment]
Think I've done all I can/should on this for now - Ended up going down a bit of a rabbit hole with our Xcode/frameworks stuff which is a bit FUBAR right now, so this just adds the tests for iOS static libraries right now and we can address the framework issues separately as there is some overlap with macOS

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
Being able to get rid of that iOS toolchain and test the install interface on iOS is promising. I've got a few more clarifying questions since idk much about iOS development.

File: .github/workflows/ci.yml
Comment (JonnyPtn) [reply .github/workflows/ci.yml:121]: Ah, they should be running - probably got lost in the crossfire when I was trying to get the framework tests running, will amend

File: examples/assets/info.plist
Comment (JonnyPtn) [reply examples/assets/info.plist:26]: I don't recall, probably just left over from when I first looked at this, will amend

File: src/SFML/Window/CMakeLists.txt
Comment (JonnyPtn) [reply src/SFML/Window/CMakeLists.txt:343]: The frameworks need to be linked by the dependent build. I believe this is propagated automatically if we use find_package along with proper targets but as we just use dumb compiler flags we have to do this (or require that anybody using it links the frameworks themselves)

File: test/install/CMakeLists.txt
Comment (JonnyPtn) [reply test/install/CMakeLists.txt:16]: Yeah as above - I've fixed the issues with the install tests (except for frameworks which is a separate can of worms) so need to enable the tests

File: test/install/CMakeLists.txt
Comment (ChrisThrasher) [inline test/install/CMakeLists.txt:23]: Why this message?

File: test/install/CMakeLists.txt
Comment (ChrisThrasher) [inline test/install/CMakeLists.txt:13]:
```suggestion
    if(IOS)
```

**Review by ChrisThrasher — APPROVED** [review]
Other than two minor comments, I like this! I can tell this is going to unblock some things I've previously tried to modernize how we handle dependencies. I'm eager to try those again once this is merged.

File: test/install/CMakeLists.txt
Comment (JonnyPtn) [reply test/install/CMakeLists.txt:23]: It was mostly useful for diagnosing SDK compatibility issues so you can see which one is being used - happy to take any direction on whether it should be removed or formatted differently

**Comment by eXpl0it3r** [comment]
Looks like, we'll need to adapt BuildBot afterwards

**Comment by eXpl0it3r** [comment]
Created a change for Build Bot, but I'm unsure if we can support this both for master and 2.6.x based branches: https://github.com/SFML/SFML-Buildbot/pull/39

**Comment by binary1248** [comment]
> Created a change for Build Bot, but I'm unsure if we can support this both for master and 2.6.x based branches: [SFML/SFML-Buildbot#39](https://github.com/SFML/SFML-Buildbot/pull/39)

This is possible... and has been implemented.

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thanks for cleaning this up! 🧹🪣

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML/blob/3f6403e279d3f0965938250f263cdd9fdfef5f4a/cmake/Macros.cmake#L35-L36

Should these lines be removed? They expand into nothing and due to how `set_property` works that results in technically valid (albeit not useful) CMake code.

**Comment by JonnyPtn** [comment]
From memory, they existed so that people could specify the properties themselves on the command line when building SFML - Can still be done with the global CMAKE_XCODE_ATTRIBUTE_... variables if these were removed, but then it wouldn't be possible to have different settings for different targets.

Benefit and cost are both very small so not sure it really matters much

**Pull request merged**
